File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
       sort(arr.begin(),arr.end());
       vector<vector<int>>ans;
       int miniDiff = INT_MAX;
       for(int i = 0; i < arr.size()-1;i++){
             miniDiff = min(arr[i+1] - arr[i],miniDiff);
       }
       for(int i = 0; i < arr.size()-1; i++){
        if(arr[i+1] - arr[i] == miniDiff){
            ans.push_back({arr[i],arr[i+1]});
        }
       }
       return ans;
    }
};

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        Arrays.sort(arr);
        List<List<Integer>> ans = new ArrayList<>();
        int mini = Integer.MAX_VALUE;
        for(int i = 0; i < arr.length-1;i++){
            int diff = arr[i+1] - arr[i];
            if(diff < mini){
                mini = diff;
            }
        }
        for(int i = 0; i < arr.length-1;i++){
            int diff = arr[i+1] - arr[i];
            if(mini == diff){
               ans.add(Arrays.asList(arr[i], arr[i + 1]));
            }
        }
        return ans;
    }
}